### PR TITLE
[Constraint] Rename Data

### PR DESCRIPTION
**Pull request by lamriaimen**
According to the guidelines

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by lamriaimen** [comment]
[ci-build][with-all-tests]

File: Sofa/Component/Constraint/Projective/src/sofa/component/constraint/projective/FixedRotationProjectiveConstraint.h [hunk 1]
```diff
@@ -65,9 +65,18 @@ class FixedRotationProjectiveConstraint : public core::behavior::ProjectiveConst
...
+
+    SOFA_ATTRIBUTE_DEPRECATED__RENAME_DATA()
+    Data< bool > FixedYRotation;
+
+    SOFA_ATTRIBUTE_DEPRECATED__RENAME_DATA()
+    Data< bool > FixedZRotation;
+
+    Data< bool > d_FixedXRotation; ///< Prevent Rotation around X axis
+    Data< bool > d_FixedYRotation; ///< Prevent Rotation around Y axis
+    Data< bool > d_FixedZRotation; ///< Prevent Rotation around Z axis
```
Comment (fredroy) [inline Sofa/Component/Constraint/Projective/src/sofa/component/constraint/projective/FixedRotationProjectiveConstraint.h]:
```suggestion
    Data< bool > d_fixedXRotation; ///< Prevent Rotation around X axis
    Data< bool > d_fixedYRotation; ///< Prevent Rotation around Y axis
    Data< bool > d_fixedZRotation; ///< Prevent Rotation around Z axis
```
The guidelines say that it should be lowerCamelCase ([N4])

File: Sofa/Component/Constraint/Projective/src/sofa/component/constraint/projective/FixedRotationProjectiveConstraint.inl [hunk 2]
```diff
@@ -33,10 +33,14 @@ namespace sofa::component::constraint::projective
...
-      FixedXRotation( initData( &FixedXRotation, false, "FixedXRotation", "Prevent Rotation around X axis")),
-      FixedYRotation( initData( &FixedYRotation, false, "FixedYRotation", "Prevent Rotation around Y axis")),
-      FixedZRotation( initData( &FixedZRotation, false, "FixedZRotation", "Prevent Rotation around Z axis"))
+      d_FixedXRotation(initData(&d_FixedXRotation, false, "FixedXRotation", "Prevent Rotation around X axis")),
+      d_FixedYRotation(initData(&d_FixedYRotation, false, "FixedYRotation", "Prevent Rotation around Y axis")),
+      d_FixedZRotation(initData(&d_FixedZRotation, false, "FixedZRotation", "Prevent Rotation around Z axis"))
 {
+          FixedXRotation.setParent(&d_FixedXRotation);
+          FixedYRotation.setParent(&d_FixedYRotation);
+          FixedZRotation.setParent(&d_FixedZRotation);
```
Comment (fredroy) [inline Sofa/Component/Constraint/Projective/src/sofa/component/constraint/projective/FixedRotationProjectiveConstraint.inl]:
```suggestion
      d_fixedXRotation(initData(&d_fixedXRotation, false, "FixedXRotation", "Prevent Rotation around X axis")),
      d_fixedYRotation(initData(&d_fixedYRotation, false, "FixedYRotation", "Prevent Rotation around Y axis")),
      d_fixedZRotation(initData(&d_fixedZRotation, false, "FixedZRotation", "Prevent Rotation around Z axis"))
{
          FixedXRotation.setParent(&d_fixedXRotation);
          FixedYRotation.setParent(&d_fixedYRotation);
          FixedZRotation.setParent(&d_fixedZRotation);
```

File: Sofa/Component/Constraint/Projective/src/sofa/component/constraint/projective/FixedRotationProjectiveConstraint.inl [hunk 3]
```diff
@@ -120,13 +124,13 @@ void FixedRotationProjectiveConstraint<DataTypes>::projectPosition(const core::M
             to_keep = twist * to_keep;
         };
 
-        if (FixedXRotation.getValue() == true){
+        if (d_FixedXRotation.getValue() == true){
```
Comment (fredroy) [inline Sofa/Component/Constraint/Projective/src/sofa/component/constraint/projective/FixedRotationProjectiveConstraint.inl]:
```suggestion
        if (d_fixedXRotation.getValue() == true){
```

File: Sofa/Component/Constraint/Projective/src/sofa/component/constraint/projective/FixedRotationProjectiveConstraint.inl [hunk 4]
```diff
@@ -120,13 +124,13 @@ void FixedRotationProjectiveConstraint<DataTypes>::projectPosition(const core::M
...
             remove_rotation(vx);
         }
-        if (FixedYRotation.getValue() == true){
+        if (d_FixedYRotation.getValue() == true){
```
Comment (fredroy) [inline Sofa/Component/Constraint/Projective/src/sofa/component/constraint/projective/FixedRotationProjectiveConstraint.inl]:
```suggestion
        if (d_fixedYRotation.getValue() == true){
```

File: Sofa/Component/Constraint/Projective/src/sofa/component/constraint/projective/FixedRotationProjectiveConstraint.inl [hunk 5]
```diff
@@ -120,13 +124,13 @@ void FixedRotationProjectiveConstraint<DataTypes>::projectPosition(const core::M
...
             remove_rotation(vy);
         }
-        if (FixedZRotation.getValue() == true){
+        if (d_FixedZRotation.getValue() == true){
```
Comment (fredroy) [inline Sofa/Component/Constraint/Projective/src/sofa/component/constraint/projective/FixedRotationProjectiveConstraint.inl]:
```suggestion
        if (d_fixedZRotation.getValue() == true){
```

**Review by fredroy — CHANGES_REQUESTED** [review]

**Review by fredroy — APPROVED** [review]

**Pull request merged**